plugins/modules/azure_rm_appgateway.py

```diff
@@ -821,11 +825,13 @@
                         default: []
             enabled:
                 description:
-                    - Whether the web application firewall is enabled or not.
+                    - (Deprecated) Whether the web application firewall is enabled or not.
```

Comment:
@l3ender You annotated these parameters, I did not find in the document to abandon the description? What documents are you referring to?

Reply:
After changes in https://github.com/ansible-collections/azure/pull/1697, trying to create an app gateway with old method results in the following error:

> Code: ApplicationGatewayWafConfigurationCannotBeAddedToWAFV2Gateway
> Message: WebApplicationFirewallConfiguration cannot be added to WAF_V2 Gateway from API version 2024-01-01. Please use a WAF Policy instead.

Nothing is mentioned for this in the [python library for azure-mgmt-network](https://pypi.org/project/azure-mgmt-network/), so this appears to be enforced on the server/API.

Reply:
This gives some info: https://learn.microsoft.com/en-us/azure/web-application-firewall/ag/create-waf-policy-ag.